scalyr_agent/util.py

```diff
@@ -561,6 +561,35 @@ def is_list_of_strings(vals):
     return True
 
 
+def get_parser_from_config(base_config, attributes, default_parser):
```

Comment:
Ok, I know you are just refactoring this code, but can you add a unit test for this function?  I'd like us to get in the habit of adding test cases when we can to help keep improving coverage.